pkg/testcase/clusterrestore.go

```diff
@@ -0,0 +1,327 @@
+package testcase
+
+import (
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/rancher/distros-test-framework/pkg/aws"
+	"github.com/rancher/distros-test-framework/pkg/customflag"
+	"github.com/rancher/distros-test-framework/shared"
+
+	. "github.com/onsi/gomega"
+)
+
+var awsConfig shared.AwsConfig
+
+func setConfigs() {
+	awsConfig = shared.AwsConfig{
+		AccessKeyID:     os.Getenv("AWS_ACCESS_KEY_ID"),
+		SecretAccessKey: os.Getenv("AWS_SECRET_ACCESS_KEY"),
+	}
+
+}
+
+func TestClusterRestoreS3(
+	cluster *shared.Cluster,
+	applyWorkload bool,
+	flags *customflag.FlagConfig,
+) {
+	setConfigs()
+
+	product := cluster.Config.Product
+	_, version, err := shared.Product()
+	Expect(err).NotTo(HaveOccurred())
+	versionStr := fmt.Sprintf("%s version ", product)
+	versionCleanUp := strings.TrimPrefix(version, versionStr)
+	endChar := strings.Index(versionCleanUp, "(")
+	versionClean := versionCleanUp[:endChar]
+
+	var workloadErr error
+	if applyWorkload {
+		workloadErr = shared.ManageWorkload("apply", product+"-extra-metadata.yaml")
+		Expect(workloadErr).NotTo(HaveOccurred(), "configmap failed to create")
+	}
+
+	shared.LogLevel("info", "%s-extra-metadata configmap successfully added", product)
+
+	testTakeS3Snapshot(
+		cluster,
+		flags,
+		true,
+	)
+	shared.LogLevel("info", "successfully completed s3 snapshot save")
+	testS3SnapshotSave(
+		cluster,
+		flags,
+	)
+	shared.LogLevel("info", "successfully validated s3 snapshot save in s3")
+	onDemandPath, onDemandPathErr := shared.FetchSnapshotOnDemandPath(cluster.Config.Product, cluster.ServerIPs[0])
+	Expect(onDemandPathErr).NotTo(HaveOccurred())
+
+	clusterToken, clusterTokenErr := shared.FetchToken(cluster.Config.Product, cluster.ServerIPs[0])
+	Expect(clusterTokenErr).NotTo(HaveOccurred())
+
+	resourceName := os.Getenv("resource_name")
+	ec2, err := aws.AddClient(cluster)
+	Expect(err).NotTo(HaveOccurred(), "error adding aws nodes: %s", err)
+
+	stopServerInstances(cluster, ec2)
+
+	stopAgentInstance(cluster, ec2)
+
+	// oldLeadServerIP := cluster.ServerIPs[0]
+
+	// create new server.
+	var serverName []string
+
+	serverName = append(serverName, fmt.Sprintf("%s-server-fresh", resourceName))
+
+	externalServerIP, _, _, createErr :=
+		ec2.CreateInstances(serverName...)
+	Expect(createErr).NotTo(HaveOccurred(), createErr)
+
+	shared.LogLevel("info", "Created server public ip: %s",
+		externalServerIP[0])
+
+	newServerIP := externalServerIP
+
+	setConfigFile(product, newServerIP[0])
+	shared.LogLevel("info", "config.yaml successfully created and copied to: /etc/rancher/%s/", product)
+
+	installProduct(
+		cluster,
+		newServerIP[0],
+		versionClean,
+	)
+	shared.LogLevel("info", "%s successfully installed on server: %s", product, newServerIP)
+
+	testRestoreS3Snapshot(
+		cluster,
+		onDemandPath,
+		clusterToken,
+		newServerIP[0],
+		flags,
+	)
+	shared.LogLevel("info", "cluster restore successful. Waiting 120 seconds for cluster "+
```

Comment:
This timeout seems too long for k3s, can you please check and see if it can be bit dynamic?